src/livekit/MediaDevicesContext.tsx

```diff
@@ -102,27 +111,39 @@ function useMediaDevice(
             // Create a virtual default audio output for browsers that don't have one.
             // Its device ID must be the empty string because that's what setSinkId
             // recognizes.
+            // we also create this if we do not have any available devices, so that
+            // we can use the default or the earpiece.
             if (
               kind === "audiooutput" &&
-              available.size &&
               !available.has("") &&
               !available.has("default")
             )
               available = new Map([
                 ["", { type: "default", name: availableRaw[0]?.label || null }],
                 ...available,
               ]);
+            if (
+              kind === "audiooutput" &&
+              (navigator.userAgent.match("iPhone") || alwaysShowIphoneEarpice)
+            )
+              // On IPhones we have to create a virtual earpiece device, because
+              // the earpiece is not available as a device ID.
+              available = new Map([
+                ...available,
+                [EARPIECE_CONFIG_ID, { type: "name", name: "Earpiece" }],
```

Comment:
This `"Earpiece"` string needs translation. Perhaps instead of using `type: "name"`, we could introduce a new `type: "earpiece"` variant to `DeviceLabel`.